doc/develop/tools/clion.rst

```diff
@@ -0,0 +1,270 @@
+
+#. Name the file ``CMakeUserPresets.json``.
+
+#. Add the following code into it:
+
+   .. code-block::
+      :emphasize-lines: 13,14
+      {
+         "version": 3,
+            "cmakeMinimumRequired": {
+            "major": 3,
+            "minor": 24
+         },
+         "configurePresets": [
+            {
+               "name": "Zephyr-Debug",
+               "generator": "Ninja",
+               "binaryDir": "build",
+               "cacheVariables": {
+                  "BOARD": "nucleo_f302r8",
+                  "CMAKE_PROGRAM_PATH": "<openocd_installation>;<home>/zephyrproject/.venv/Scripts"
+               }
+            }
+         ],
+         "buildPresets": [
+            {
+               "name": "Zephyr-Debug",
+               "configurePreset": "Zephyr-Debug"
+            }
+         ]
+      }
+
+#. Customize the following lines:
+
+   * ``"BOARD":...`` - specify the board you are using.
+
+   * ``"CMAKE_PROGRAM_PATH":...`` - set the path to Zephyr environment scripts
+     (delete this part if you are not using a virtual environment) and set
+     the path to your flashing software (OpenOCD in our example).
+
+#. Save the file. Make sure it is located in the ``hello_world`` folder,
+    i.e. the same folder as your CMake project.
+
+Load CMake
+**********
+
+#. In the **Project** tree, locate the ``CMakeLists.txt`` file in the
+    ``hello_world`` folder.
+
+#. Right-click it and select **Load CMake Project**.
+
+   CMake load should finish successfully.
+
+Configure CMake profiles
+************************
+
+CLion automatically creates a CMake profile for any detected preset,
+but such profiles are disabled by default. Follow the steps below to
+enable the profile associated with our preset and set it as a working
+profile for the project.
+
+#. Go to **Settings | Build, Execution, Deployment | CMake**.
+
+#. Select the **Zephyr-Debug preset** profile and set the **Enable profile**
+   checkbox.
+
+#. Delete the automatically created **Debug** profile. Alternatively,
+    move the **Zephyr-Debug preset** profile to the top of the list
+    to make it default for the project.
+
+#. Click **Apply** and close the dialog.
+
+   CMake project will be reloaded.
+
+Build the project
+*****************
+
+#. Check the top toolbar: the **Zephyr-Debug** profile and the
+   **zephyr_final** configuration should be selected.
+
+#. Click the hammer icon next to the configuration switcher.
+
+   When the build is finished, you can find build artifacts in
+   the ``hello_world\build`` folder.
+
+Note that other CMake targets like ``puncover`` or ``hardenconfig``
+can also be called at this point.
+
+Configure Zephyr parameters for debug
+*************************************
+
+#. In the configuration switcher, select **guiconfig** and click the hammer
+   icon.
+
+#. Use the GUI application to set the following flags:
+
+   .. code-block::
+
+      CONFIG_DEBUG_THREAD_INFO
```

Comment:
Atcually you need to remove the CONFIG_ to search for it.

Reply:
@kartben @lucsegers Thank you! Sorry I missed that in the first pass

Reply:
No worries! I have started another (I think final!) round of reviews and I'm pretty confident we can get this in by the end of the year 🙃 thanks again for all the efforts on this one